File: python/gpt/p109.py
"""
p109.py
Read /etc/passwd and print lines matching username provided as argv[1]
"""
import sys

if len(sys.argv) < 2:
    print('usage: p109.py username')
    sys.exit(1)
user = sys.argv[1]
with open('/etc/passwd', 'r', encoding='utf-8') as f:
    for line in f:
        if user in line:
            print(line, end='')

File: c/gpt/p109.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>

/* p109.c
   Read username from argv and search /etc/passwd for matching lines.
*/
int main(int argc, char **argv){
    if (argc < 2) { fprintf(stderr, "usage: %s username\n", argv[0]); return 1; }
    const char *user = argv[1];
    FILE *f = fopen("/etc/passwd","r");
    if (!f) { perror("fopen"); return 1; }
    char line[1024];
    while (fgets(line, sizeof(line), f)){
        if (strstr(line, user)) fputs(line, stdout);
    }
    fclose(f);
    return 0;
}
